invokeai/app/invocations/bria_text_encoder.py

```diff
@@ -0,0 +1,93 @@
+from typing import Optional
+
+import torch
+from transformers import (
+    T5EncoderModel,
+    T5TokenizerFast,
+)
+
+from invokeai.app.invocations.model import T5EncoderField
+from invokeai.app.invocations.primitives import BaseInvocationOutput, FieldDescriptions, Input, OutputField
+from invokeai.app.services.shared.invocation_context import InvocationContext
+from invokeai.backend.bria.pipeline_bria_controlnet import encode_prompt
+from invokeai.invocation_api import (
+    BaseInvocation,
+    Classification,
+    InputField,
+    LatentsField,
+    invocation,
+    invocation_output,
+)
+
+
+@invocation_output("bria_text_encoder_output")
+class BriaTextEncoderInvocationOutput(BaseInvocationOutput):
+    """Base class for nodes that output a CogView text conditioning tensor."""
+
+    pos_embeds: LatentsField = OutputField(description=FieldDescriptions.cond)
+    neg_embeds: LatentsField = OutputField(description=FieldDescriptions.cond)
+    text_ids: LatentsField = OutputField(description=FieldDescriptions.cond)
+
+
+@invocation(
+    "bria_text_encoder",
+    title="Prompt - Bria",
+    tags=["prompt", "conditioning", "bria"],
+    category="conditioning",
+    version="1.0.0",
+    classification=Classification.Prototype,
+)
+class BriaTextEncoderInvocation(BaseInvocation):
+    prompt: str = InputField(
+        title="Prompt",
+        description="The prompt to encode",
+    )
+    negative_prompt: Optional[str] = InputField(
+        title="Negative Prompt",
+        description="The negative prompt to encode",
+        default="Logo,Watermark,Text,Ugly,Morbid,Extra fingers,Poorly drawn hands,Mutation,Blurry,Extra limbs,Gross proportions,Missing arms,Mutated hands,Long neck,Duplicate",
+    )
+    max_length: int = InputField(
+        default=128,
+        title="Max Length",
+        description="The maximum length of the prompt",
+    )
```

Comment:
Is this intended to be user-configurable? Are there and constraints on the value? If so, please use pydantic to enforce them. 

For FLUX, we allow only two values. We indicate this with `t5_max_seq_len: Literal[256, 512]`.